Rename path type to recursive relationship (#217)
kuzudb/kuzu-docs: src/content/docs/cypher/expressions/path-functions.md

---
title: Recursive relationship functions

> description: Recursive relationship functions are used to manipulate a recursive relationship.

Reviewer: ```suggestion
description: Functions that are used to manipulate recursive relationships
```

---

| Function | Description |
| ----------- | ----------- |
| `NODES`| returns all nodes from a recursive relationship |
| `RELS` | returns all rels from a recursive relationship |
| `PROPERTIES` | return given property from nodes/relationships |

> | `IS_TRAIL` | check if path contains repeated relelationships |

Reviewer: ```suggestion
| `IS_TRAIL` | check if path contains repeated relationships |
```

| Input type | Output type |
| ----------- | ----------- |

> | `RECURSIVE RELATIONSHIP` | `LIST[NODE]` |

Reviewer: ```suggestion
| `RECURSIVE_REL` | `LIST[NODE]` |
```

```
### RELS

> Returns all relationships from path.

Reviewer: ```suggestion
Returns all relationships from a recursive relationship.
```

| Input type | Output type |
| ----------- | ----------- |

> | `RECURSIVE RELATIONSHIP` | `LIST[RELELATIONSHIP]` |

Reviewer: ```suggestion
| `RECURSIVE_REL` | `LIST[REL]` |
```

### PROPERTIES

> Return given property from nodes/relelationships.

Reviewer: ```suggestion
Return given property from nodes/relationships.
```

| Input type | Output type |
| ----------- | ----------- |

> | `LIST[NODE/RELATIONSHIP]`, `STRING` | `LIST[ANY]` |

Reviewer: ```suggestion
| `LIST[NODE/REL]`, `STRING` | `LIST[ANY]` |
```

### IS_TRAIL

Check if a recursive relationship contains repeated relationships.

| Input type | Output type |
| ----------- | ----------- |

> | `RECURSIVE RELATIONSHIP` | `BOOLEAN` |

Reviewer: ```suggestion
| `RECURSIVE_REL` | `BOOLEAN` |
```

### IS_ACYCLIC

Check if a recursive relationship contains repeated nodes.

| Input type | Output type |
| ----------- | ----------- |

> | `RECURSIVE RELATIONSHIP` | `BOOLEAN` |

Reviewer: ```suggestion
| `RECURSIVE_REL` | `BOOLEAN` |
```
